### DISCUSSION TITLE
Quicklook for .Rmd files

### DISCUSSION THREAD

**Discussion by ajcullum** [sbarex/QLMarkdown#61] [Ideas]
The statistical programing language/system R supports markdown files that include a few lines of additional code at the top aimed at R, but are otherwise largely standard markdown. They use the extension Rmd rather than just md. Would it be possible to get that extension added to the list QLMarkdown looks for? Or alternatively, can someone point me to the specific QLMarkdown file(s) that could be altered locally to support that? By default they're not even recognized as text files by macOS, and It would be really handy to be able to see the contents when browsing a directory. 

Thanks,
AJC

**Comment by sbarex** [comment]
I don't know `rmarkdown`. I understand that he combines `markdown` syntax with `yaml` and portions of `r` code that are evaluated at render time.

The simplest solution would be to render the `.rmd` files as pure `.md`. But the header of the file (which I think follows the syntax of `yaml`), however, I am afraid it may be misinterpreted by the markdown parser. 

Before making this code change, you might try renaming some files with the `.md` extension to see if the formatting might still work. 

If that doesn't work I might consider creating an extension that processes at least the `.rmd` file header but not the inline execution of the `r` statements. To do this, however, I would need a minimum of (simple) documentation to understand how the `.rmd` file is organized.

**Reply by ajcullum** [reply]
Thanks so much for considering it! First, I absolutely wouldn't be expecting any R execution. The main thing I'm looking for is the ability to read the contents. 

Here's some additional information that might be helpful. At least for the basic Rmd files I make, the headers are fairly simple. These would be some typical examples. These appear at the start of the file:

```
---
title: "Goodness-of-Fit Analysis in R – Part 1"
author: "Alistair Cullum"
output: html_notebook
---
```
or
```
---
title: "Goodness-of-Fit Analysis in R – Part 1"
author: "Alistair Cullum"
output:
  pdf_document: default
  html_notebook: default
---
```

For that second version, here's how it renders in QLMarkdown:


![image](https://user-images.githubusercontent.com/36862784/109664563-deb64300-7b32-11eb-928d-0264dc70ddf0.png)

So other than the fact that there are no line breaks, it's not bad. I don't know about anyone else, but I'd be perfectly happy with that version if it let me see the rest of the file. There is a slightly better rendering in MacDown, which treats it as a table, but it's not a major concern for me:

![image](https://user-images.githubusercontent.com/36862784/109667204-8765a200-7b35-11eb-988a-4b38fb6565e3.png)


Here's some additional information on the headers: https://richardjacton.github.io/Advanced-Rmarkdown-YAML-headers/ 

If I were looking for an easy solution to formally handle the headers in a file with the Rmd extension, I might just treat that material between the first two `---` marks as a code block. 

Thanks again,
AJC

**Reply by sbarex** [reply]
Thanks for the info.

Please send me the output of this command launched in a terminal shell (replace PATH_OF_A_RMD_FILE with the path of an exists .rmd file):
```
 mdls -name kMDItemContentType -name kMDItemContentTypeTree -name kMDItemContentType PATH_OF_A_RMD_FILE
```

**Reply by ajcullum** [reply]
Here's the output:
```
kMDItemContentType     = "dyn.ah62d4rv4ge81e5pe"
kMDItemContentType     = "dyn.ah62d4rv4ge81e5pe"
kMDItemContentTypeTree = (
    "public.item",
    "dyn.ah62d4rv4ge81e5pe",
    "public.data"
)
```

**Comment by sbarex** [comment]
Try the new release (b24) where I added support for `.rmd` files.

**Reply by ajcullum** [reply]
That works a treat! Thanks so much for doing that! If you have a tip jar system somewhere, or some group you'd like me to  donate to, I'd be happy to show my appreciation for your work.

**Reply by sbarex** [reply]
Thank you very much, for the moment it is a job that I do in my free time to distract myself a little, but I am very pleased that it can also be useful to others!

In this difficult period, if you want, I'm sure you can find some projects to donate to.

Thank you so much for your kindness
